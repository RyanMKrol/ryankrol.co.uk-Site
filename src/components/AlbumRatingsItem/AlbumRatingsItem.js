import React, { Component } from 'react';

import styles from './AlbumRatingsItem.module.css';

/**
 * A component representing a movie ratings item
 */
class AlbumRatingsItem extends Component {
  /**
   * @class
   * @param {object} props arguments to class
   */
  constructor(props) {
    super();

    if (
      typeof props.artist === 'undefined'
      || typeof props.date === 'undefined'
      || typeof props.highlights === 'undefined'
      || typeof props.mood === 'undefined'
      || typeof props.rating === 'undefined'
      || typeof props.title === 'undefined'
      || typeof props.thumbnail === 'undefined'
    ) {
      throw new Error('Could not create AlbumRatingsItem');
    }

    this.artist = props.artist;
    this.date = props.date;
    this.highlights = props.highlights;
    this.mood = props.mood;
    this.rating = props.rating;
    this.title = props.title;
    this.thumbnail = props.thumbnail;
  }

  /**
   * Render method
   *
   * @returns {React.Component} JSX AlbumRatingsItem component
   */
  render() {
    return (
      <div className={styles.outerContainer}>
        <div className={styles.container}>
          <div className={styles.titleContainer}>
            <div className={styles.innerTitleContainer}>
              <p className={styles.titleContent}>
                {this.artist} - {this.title}
              </p>
              <div className={styles.titleUnderline} />
            </div>
          </div>
          <div className={styles.content}>
            <div className={styles.thumbnailContainer}>
              <img className={styles.thumbnail} src={this.thumbnail} alt={this.title} />
            </div>
            <div>
              <span>Score: </span>
              <span>{this.rating}</span>
              <br />
              <span>Mood: </span>
              <span>{this.mood}</span>
              <br />
              <br />
              <span>Highlights: </span>
              <span>{this.highlights}</span>
              <br />
            </div>
          </div>
          <p className={styles.date}>{this.date}</p>
        </div>
      </div>
    );
  }
}

export default AlbumRatingsItem;
