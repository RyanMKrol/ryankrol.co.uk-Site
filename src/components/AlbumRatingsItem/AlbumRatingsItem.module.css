.outerContainer {
  padding: 4px;
}

.container {
  display: flex;
  justify-content: center;

  width: 500px;

  background-color: rgba(255, 255, 255, 0.9);
  flex-direction: column;

  border-radius: 4px;
}

.titleContainer {
  display: flex;
  flex-direction: column;
  width: 100%;
  align-items: center;
}

.titleContent {
  margin-bottom: 4px;
  text-align: center;
}

.titleUnderline {
  border-top-style: solid;
  border-top-color: black;
  border-top-width: 1px;
  width: 100%;
  height: auto;
}

.content {
  display: flex;
  align-items: center;
  font-size: 12px;
  padding: 4px;
}

.thumbnailContainer {
  justify-content: center;
  align-items: center;
  max-height: 160px;
  padding: 8px;
}

.thumbnail {
  max-height: 160px;
  border-radius: 4px;
}

.date {
  margin: 4px;
  font-size: 10px;
  width: 100%;
  text-align: center;
}

@media (max-width: 580px) {
  .thumbnailContainer {
    max-height: 120px;
  }

  .thumbnail {
    max-height: 120px;
  }

  .container {
    width: 410px;
  }
}

@media (max-width: 420px) {
  .container {
    width: 320px;
  }
}
