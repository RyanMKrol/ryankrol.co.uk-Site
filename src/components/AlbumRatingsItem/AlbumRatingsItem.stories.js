import React from 'react';
import AlbumRatingsItem from './AlbumRatingsItem';

export default {
  title: 'Components/AlbumRatingsItem',
};

/**
 * default story
 *
 * @returns {string} JSX for the component I want to display
 */
export const normal = () => (
  <div>
    <AlbumRatingsItem
      artist={'Gorillaz'}
      title={'Demon Days'}
      date={'15-03-21'}
      highlights={'Dirty Harry, Feel Good Inc.'}
      mood={'Upbeat'}
      rating={84}
    />
  </div>
);
