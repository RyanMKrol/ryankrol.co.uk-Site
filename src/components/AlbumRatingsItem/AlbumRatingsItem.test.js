import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import AlbumRatingsItem from './AlbumRatingsItem';

const defaultArgs = {
  artist: 'Gorillaz',
  title: 'Demon Days',
  date: '15-03-21',
  highlights: 'Dirty Harry, Feel Good Inc.',
  mood: 'Upbeat',
  rating: 84,
};

// Render Tests

it('renders AlbumRatingsItem correctly', () => {
  const albumRatingsItem = shallow(<AlbumRatingsItem {...defaultArgs} />);
  expect(toJson(albumRatingsItem)).toMatchSnapshot();
});

it('throws when required parameters are omitted', () => {
  expect(() => mount(<AlbumRatingsItem />)).toThrow();
});

it('does not throw when required parameters are provided', () => {
  expect(() => mount(<AlbumRatingsItem {...defaultArgs} />)).not.toThrow();
});
