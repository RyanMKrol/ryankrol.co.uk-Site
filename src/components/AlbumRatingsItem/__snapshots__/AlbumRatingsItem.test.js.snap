// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AlbumRatingsItem without issue 1`] = `
<div
  className="outerContainer"
>
  <div
    className="container"
  >
    <div
      className="titleContainer"
    >
      <div
        className="innerTitleContainer"
      >
        <p
          className="titleContent"
        >
          Gorillaz
           - 
          Demon Days
        </p>
        <div
          className="titleUnderline"
        />
      </div>
    </div>
    <div
      className="content"
    >
      <div
        className="thumbnailContainer"
      >
        <img
          alt="Demon Days"
          className="thumbnail"
          src="https://lastfm.freetls.fastly.net/i/u/300x300/271483e955d2b255160f3361a7f5fb78.png"
        />
      </div>
      <div>
        <span>
          Score: 
        </span>
        <span>
          84
        </span>
        <br />
        <span>
          Mood: 
        </span>
        <span>
          Upbeat
        </span>
        <br />
        <br />
        <span>
          Highlights: 
        </span>
        <span>
          Dirty Harry, Feel Good Inc., White Light, Dare, Fire Coming Out of the Monkey's Head
        </span>
        <br />
      </div>
    </div>
    <p
      className="date"
    >
      15-03-21
    </p>
  </div>
</div>
`;
